### PR TITLE
docs(General): Add JSDoc on all exported functions and types

### PR DESCRIPTION
**Pull request by bboure**

### CODE REVIEW

File: src/utils/cognito.ts [hunk 1]
```diff
@@ -79,20 +113,35 @@ export const cognitoSignUp = async (params: {
...
+   */
+  username: string;
+  /**
+   * An optional cognito client configuration.
+   */
+  clientConfig?: CognitoIdentityProviderClientConfig;
+} & CognitoInput;
+
 /**
  * Delete a user in cognito.
```
Comment (bayoudhi) [inline src/utils/cognito.ts]:
```suggestion
 * Delete a user from cognito.
```

File: src/utils/cognito.ts [hunk 2]
```diff
@@ -79,20 +113,35 @@ export const cognitoSignUp = async (params: {
     userPoolId,
     password,
     username,
-    config,
+    clientConfig,
   });
 };
 
+export type DeleteCognitoUserInput = {
```
Comment (bayoudhi) [inline src/utils/cognito.ts:123]: Should this be `CognitoDeleteUserInput` to follow the same naming convention (`CognitoSignUpInput` & `CognitoSignInInput`)?

Also, we probably want to add JSDoc comments for this type

**Review by bayoudhi — APPROVED** [review]
Great work! I left two minor comments. 

As you mentioned to me earlier this morning, it appears that we should establish a more consistent naming convention across various elements (matchers, spies, helpers, etc...) Let's work on this this next :)

File: src/utils/cognito.ts (hunk 2 above)
Comment (bboure) [reply src/utils/cognito.ts:123]: good catch. in fact I meant to rename the function :)
I think `deleteCognitoUser()` sounds better than `cognitoDeleteUser()` wdyt?

Comment (bayoudhi) [reply src/utils/cognito.ts:123]: Sounds good, should we also rename the other two functions 👇🏻  to be more consistent?

`cognitoSignUp` --> `signUpCognitoUser`
`cognitoSignIn` --> `signInCognitoUser`

Comment (bboure) [reply src/utils/cognito.ts:123]: I went with 
`signUpCognitoUser` (sign up a user)
 and 
`signInWithCognitoUser` (sign in **with** a user)

**Review by bayoudhi — APPROVED** [review]

**Pull request merged**